docs/src/main/sphinx/sql/match-recognize.rst

```diff
@@ -0,0 +1,575 @@
+===============
+MATCH_RECOGNIZE
+===============
+
+Synopsis
+--------
+
+.. code-block:: text
+
+    MATCH_RECOGNIZE (
+      [ PARTITION BY column [, ...] ]
+      [ ORDER BY column [, ...] ]
+      [ MEASURES measure_definition [, ...] ]
+      [ rows_per_match ]
+      [ AFTER MATCH skip_to ]
+      PATTERN ( row_pattern )
+      [ SUBSET subset_definition [, ...] ]
+      DEFINE variable_definition [, ...]
+      )
+
+Description
+-----------
+
+The ``MATCH_RECOGNIZE`` clause is used to detect patterns across rows of a table. Patterns of interest are
+specified using row pattern syntax based on regular expressions. When the pattern is detected, requested
+information, specified as the row pattern measures, is retrieved from rows that match certain components
```

Comment:
This sentence is confusing and vague. I am not sure what you are trying to say exactly. Maybe this:

One or more rows are returned, when a match to the pattern is detected.